src/main/java/play/dev/filewatch/FileWatchService.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package play.dev.filewatch;
+
+import java.io.File;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+/** A service that can watch files */
+public interface FileWatchService {
+
+  /**
+   * Watch the given sequence of files or directories.
+   *
+   * @param filesToWatch The files to watch.
+   * @param onChange A callback that is executed whenever something changes.
+   * @return A watcher
+   */
+  FileWatcher watch(Iterable<File> filesToWatch, Runnable onChange);
+
+  /** @deprecated Use {@link #watch(Iterable, Runnable)} instead */
+  @Deprecated(since = "3.0.0", forRemoval = true)
+  default FileWatcher watch(List<File> filesToWatch, Callable<Void> onChange) {
+    return watch(filesToWatch, () -> onChange.call());
+  }
+
+  /**
+   * @deprecated Renamed to {@link #detect(int, LoggerProxy)}, the first argument targetDirectory of
+   *     type File is not used anymore
+   */
+  @Deprecated(since = "3.0.0", forRemoval = true)
+  static FileWatchService defaultWatchService(
+      File targetDirectory, int pollDelayMillis, LoggerProxy logger) {
+    return detect(pollDelayMillis, logger, false);
+  }
+
+  /**
+   * @deprecated Renamed to {@link #detect(int, LoggerProxy, Boolean)}, the first argument
+   *     targetDirectory of type File is not used anymore
+   */
+  @Deprecated(since = "3.0.0", forRemoval = true)
+  static FileWatchService defaultWatchService(
+      File targetDirectory, int pollDelayMillis, LoggerProxy logger, Boolean disableFileHashCheck) {
+    return detect(pollDelayMillis, logger, disableFileHashCheck);
+  }
```

Comment:
I specially want to keep this method, because I am 100% sure Play users currently manually override the [`fileWatchService` config](https://github.com/playframework/playframework/blob/18b6720d6a395f6a94b47804a3db5368cada44ef/dev-mode/sbt-plugin/src/main/scala/play/sbt/PlaySettings.scala#L144-L146) in their Play projects to disable file hash checking. See
- #111